pipenv/core.py

```diff
@@ -2730,6 +2730,9 @@ def do_run(
     """
     from .cmdparse import ScriptEmptyError
 
+    load_dot_env(project, quiet=quiet)
+    env = os.environ.copy()
+
```

Comment:
There was a past discussion about when we should parse dotenv. I hope this doesn't break any user expectations.

Reply:
Oh I need to move this to a different PR.   This change was because right now it does not work to set PIPENV_CUSTOM_VENV_NAME in the .env file because it gets loaded too late.